File: Quivi.Backoffice.Api/Requests/ModifierGroups/PatchModifierGroupRequest.cs
using Quivi.Domain.Entities;

namespace Quivi.Backoffice.Api.Requests.ModifierGroups
{
    public class PatchModifierGroupRequest : ARequest
    {
        public string? Name { get; init; }
        public int? MinSelection { get; init; }
        public int? MaxSelection { get; init; }
        public IDictionary<string, PatchModifierItem>? Items { get; init; }
        public IDictionary<Language, PatchModifierGroupLanguage>? Translations { get; init; }
    }

    public class PatchModifierItem
    {
        public decimal? Price { get; init; }
        public int? SortIndex { get; init; }
    }

    public class PatchModifierGroupLanguage
    {
        public string? Name { get; init; }
    }
}

File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/modifierGroups/PatchModifierGroupRequest.ts
import { Language } from "../Language";

export interface PatchModifierGroupRequest {
    readonly id: string;
    readonly name?: string;
    readonly minSelection?: number;
    readonly maxSelection?: number;
    readonly items?: Record<string, PatchModifierItem | undefined>;
    readonly translations?: Record<Language, PatchModifierGroupTranslation> | undefined;
}

export interface PatchModifierItem {
    readonly price?: number;
    readonly sortIndex?: number;
}

export interface PatchModifierGroupTranslation {
    readonly name?: string;
}